heron/packing/src/java/com/twitter/heron/packing/roundrobin/ResourceCompliantRRPacking.java

```diff
@@ -153,7 +163,45 @@ public PackingPlan pack() {
     Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
 
     Set<PackingPlan.ContainerPlan> containerPlans = PackingUtils.buildContainerPlans(
-        containerInstances, ramMap, this.defaultInstanceResources, paddingPercentage);
+        resourceCompliantRRAllocation.get(), ramMap, this.defaultInstanceResources,
+        paddingPercentage);
+    return new PackingPlan(topology.getId(), containerPlans);
+  }
+
+  /**
+   * Get a new packing plan given an existing packing plan and component-level changes.
+   *
+   * @return new packing plan
+   */
+  public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges) {
+    int adjustments = 0;
+    this.numAdjustments = 0;
+    this.numContainers = currentPackingPlan.getContainers().size();
+    this.containerId = 1;
+
+    int additionalContainers = computeNumAdditionalContainers(componentChanges, currentPackingPlan);
+    increaseNumContainers(additionalContainers);
```

Comment:
@billonahill In the algo I only need the increaseNumContainers method. There is no place in the code that does a reset. Even if I had to use reset, I would still compute the additionalContainers and then add them to the current containers size and then invoke reset. Am I missing sth here?